lib/OpenQA/Parser/Format/JUnit.pm

```diff
@@ -94,11 +99,17 @@ sub parse {
                 $tc_result =~ s/^skipped$/missing/;
                 $tc_result =~ s/^error$/unknown/;     # error in the testsuite itself
                 $tc_result =~ s/^failure$/fail/;      # test failed
+                $tc_result =~ s/^softfail.*$/softfail/; # test softfailed
             }
 
-            $result->{result} = 'fail' if $tc_result eq 'fail';
+            if ($tc_result eq 'fail') {
+                $result->{result} = 'fail';
+	    }
+	    elsif ($tc_result eq 'softfail' && $result->{result} ne 'fail') {
+		$result->{result} = 'softfail';
+	    }
 
-            my $details = {result => $tc_result};
+	    my $details = {result => $tc_result};
```

Comment:
We don't use tabs. Please run `tools/tidy`. The CI checks have failed anyways so it is useful to re-trigger it anyways.

Reply:
Done